src/lib/mina_metrics/no_metrics/mina_metrics.ml

```diff
@@ -59,6 +59,36 @@ module Runtime = struct
   let long_async_job : Long_job_histogram.t = ()
 end
 
+module Process_memory = struct
+  let rss_update_interval_mins : float ref = ref 1.
+
+  module type Rss_gauge_intf = sig
+    val set_pid : int -> unit
+
+    val clear_pid : unit -> unit
+  end
+
+  module Make_rss_gauge () : Rss_gauge_intf = struct
+    let set_pid _ = ()
+
+    let clear_pid () = ()
+  end
+
+  module Daemon = Make_rss_gauge ()
+
+  module Prover = Make_rss_gauge ()
+
+  module Verifier = Make_rss_gauge ()
+
+  module Snark_worker = Make_rss_gauge ()
```

Comment:
I suggest we add a note here that this only tracks memory used by subprocess snark workers. Since there's snark worker that aren't spawned as a subprocess.